/*!
* @file
* @brief IQuantumPath
*
* @copyright Copyright (C) University of Bristol 2017
*    This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
*    If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
*    See LICENSE file for details.
* @date 23/01/2018
* @author Richard Collins <richard.collins@bristol.ac.uk>
*/
syntax = "proto3";
import public "google/protobuf/empty.proto";
import public "Reporting.proto";
import public "Site.proto";

package cqp.remote;

/**
 * @brief Communitcate with the network manager which will decide the topology of the QKD network
 */
service INetworkManager {

    /**
    * @brief RegisterSite
    *   Register a site with the Quantum path algorithm, detailing how the site can be connected
    *  This mat be re-called to update the statea of an already registered site.
    * @param request The details of the site registering
    * @return status
    */
    rpc RegisterSite(Site) returns (google.protobuf.Empty);

    /**
    * @brief UnregisterSite
    *   Notify the server that the site can no longer be contacted.
    * @param request The address of the site to unregister
    * @return status
    */
    rpc UnregisterSite(SiteAddress) returns (google.protobuf.Empty);

    rpc GetRegisteredSites(google.protobuf.Empty) returns (SiteDetailsList);
}
